### PR TITLE
Add warnings handler

### PR DESCRIPTION
**Pull request by jaclarke**
See https://github.com/edgedb/edgedb/issues/7822 for details.

### CODE REVIEW

File: packages/driver/src/baseConn.ts [hunk 1]
```diff
@@ -1230,7 +1272,7 @@ export class BaseRawConnection {
         (!inCodec && args !== null) ||
         (this.stateCodec === INVALID_CODEC && state !== Session.defaults())
       ) {
-        [card, inCodec, outCodec] = await this._parse(
+        [card, inCodec, outCodec, _, _, _, warnings] = await this._parse(
```
Comment (scotttrinh) [inline packages/driver/src/baseConn.ts:1275]: btw, I _think_ these `_` variable names are unnecessary? Maybe eslint would complain if you did this, but I think this is equivalent:

```suggestion
        [card, inCodec, outCodec,,,, warnings] = await this._parse(
```

Just a tiny nit, not even something I'm suggesting we do here, just putting it out there.

File: packages/driver/src/errors/base.ts [hunk 2]
```diff
@@ -34,17 +35,15 @@ export class EdgeDBError extends Error {
     return this.constructor.name;
   }
 
-  hasTag(tag: symbol): boolean {
-    // Can't index by symbol, except when using <any>:
-    //   https://github.com/microsoft/TypeScript/issues/1863
-    const error_type = this.constructor as typeof EdgeDBError as any;
-    return Boolean(error_type.tags?.[tag]);
+  hasTag(tag: tags): boolean {
+    const error_type = this.constructor as typeof EdgeDBError;
```
Comment (scotttrinh) [inline packages/driver/src/errors/base.ts:39]: 🎉

File: packages/driver/test/client.test.ts [hunk 3]
```diff
@@ -2080,6 +2083,43 @@ test("pretty error message", async () => {
...
+test("warnings handler", async () => {
+  if (getEdgeDBVersion().major < 6) return;
+
+  let client = getClient();
+
+  try {
+    let warnings: EdgeDBError[] | null = null;
+    client = client.withWarningHandler((_warnings) => (warnings = _warnings));
+
+    await expect(client.query("select _warn_on_call();")).resolves.toEqual([0]);
```
Comment (scotttrinh) [inline packages/driver/test/client.test.ts:2095]: This is some kind of built-in?

**Review by scotttrinh — APPROVED** [review]
👑

File: packages/driver/src/baseConn.ts (hunk 1 above)
Comment (jaclarke) [reply packages/driver/src/baseConn.ts:1275]: Huh, I didn't know you could do that 🌈⭐. Though I think I prefer the underscores as it looks more intentional to me, and not that I've just typoed it.

File: packages/driver/test/client.test.ts (hunk 3 above)
Comment (jaclarke) [reply packages/driver/test/client.test.ts:2095]: Yes, Sully 'documented' it here: https://github.com/edgedb/edgedb/pull/7823. I think it only exists in test mode/dev mode though.

**Pull request merged**